### PR TITLE
Flag answers as spam (closes #53)

### LINKED ISSUE #53: API: Mark an Answer as Spam
As a developer I want to have an option to mark and unmark an answer as spam.

### PR DESCRIPTION
**Pull request by zsoltime**
I haven't added tests yet as I'm waiting for my other PR (#68) to be accepted first.

### CODE REVIEW

File: server/models/answer.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ const answerSchema = mongoose.Schema(
     question_id: { type: mongoose.Schema.Types.ObjectId, required: true },
     body: { type: String, required: true },
     votes: { type: Number, default: 0 },
+    flagged_by: { type: Array, default: [] },
```
Comment (faityworld) [inline server/models/answer.js]: The default is not necessary. We should use it for set up default values. But the array is empty.

**Review by faityworld — CHANGES_REQUESTED** [review]

File: server/models/answer.js (hunk 1 above)
Comment (zsoltime) [reply server/models/answer.js]: You're right, I've removed that. Thanks :)

**Comment by zsoltime** [comment]
And finally added the tests too 😃

**Review by faityworld — APPROVED** [review]
Looks good

**Pull request merged**